### ISSUE TITLE
dataset thumbnail descriptor leak

### ISSUE THREAD

**Issue by pameyer** [IQSS/dataverse#5017]
Identified in minor fork from d4860b467e6c711ef58e9d88d29dfb2312208dc6 (aka - 5004-respect_pub_args before it got merged) with `wget` (full crawl of a staging site); CentOS 7, POSIX storage.  Read descriptors not closed for `dataset_logo.thumb48`, amplification factor (if any) unknown.

Needs additional investigation (reproduce on develop, ideally identify URL that can be used with curl for dev/QA).

**Comment by pameyer** [comment]
reproducible in docker-aio w\ setupIT (`wget --no-check-certificate -e robots=off -m --reject-regex="q=" ${DATAVERSE_HOST}`; `lsof | grep -c thumb`) on same fork; fix for read descriptor leak tentatively identified. Appears to be a separate write descriptor leak as well.  Tentative fix resolves leak with setupIT/docker-aio, does not resolve leak on staging site (probably due to "non-dataset" thumbnail).  Additional investigation sorted it out in the "non-dataset thumbnail" case as well.

2nd time that there's been a resource leak with the pattern `if (null != do_something_that_opens_a_resources(){ x=do_something_that_opens_a_resources();}` .  This pattern appears to not be detected by static analyzers I've used; but if it happens 2x it strongly suggests that it's worth seeing where else in the codebase this pattern is used.

For future reference, leak appears to be from the `data:image/png;base64` encoded thumbnail, rather than a direct URL.

Resolved on fork; wil backport to develop for PR
